src/formatDate.js

```diff
@@ -50,7 +50,45 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = toFormat[3];
+  const dateToArray = date.split(fromFormat[3]);
+  const fromDates = {};
+  const toDates = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    fromDates[fromFormat[i]] = dateToArray[i];
+    toDates[toFormat[i]] = 0;
+  }
+
+  let keyYearFrom = '';
+
+  for (const key in fromDates) {
+    if (key.includes('YY')) {
+      keyYearFrom = key;
+    }
+  }
+
+  let keyYearTo = '';
+
+  for (const key in toDates) {
+    if (key.includes('YY')) {
+      keyYearTo = key;
+    }
+  }
+
+  if (keyYearFrom.length > keyYearTo.length) {
+    fromDates[keyYearTo] = fromDates[keyYearFrom].toString().slice(2);
+  }
+
+  if (keyYearFrom.length < keyYearTo.length && fromDates[keyYearFrom] < 30) {
+    fromDates[keyYearTo] = `20${fromDates[keyYearFrom]}`;
+  }
+
+  if (keyYearFrom.length < keyYearTo.length && fromDates[keyYearFrom] >= 30) {
+    fromDates[keyYearTo] = `19${fromDates[keyYearFrom]}`;
+  }
```

Comment:
we can update our condition and use the ternary operator instead of the two `if`s

Reply:
✅